api/src/main/java/jakarta/servlet/ServletContext.java

```diff
@@ -1436,4 +1517,87 @@
+     */
+    public enum PathCodingMode {
+        /**
+         * Effected methods will return/accept paths as was done prior to the introduction of the PathCodingMode in Servlet 6.0.
+         * The path returned by {@link HttpServletRequest#getContextPath()} is not decoded. Paths returned by the methods
+         * {@link HttpServletRequest#getServletPath()} and {@link HttpServletRequest#getPathInfo()} are fully decoded and have
+         * all path parameters removed. Decoding may include ambiguous characters such as a decoded <code>%2F</code>, which is
+         * indistinguishable from a segment separator '/' in the request URI. Methods accepting paths may accept encoded
+         * characters. A '?' character in a path will be treated as indicating a query string and will not be encoded. Some
+         * implementation specific behavioural differences may be apparent in this mode.
+         */
+        LEGACY,
+
+        /**
+         * Effected methods will only return/accept paths that do not contain any encoded reserved characters and with all path
+         * parameters removed. If the requested URI contains encoded reserved characters, then a call to a method that returns
+         * such a path will throw {@link IllegalStateException}. If a path with encoded reserve characters is passed to a method
+         * taking a path as a parameter, then it will throw {@link MalformedURLException} if declared. otherwise a
+         * {@link IllegalArgumentException}.
+         */
+        DECODED,
+
+        /**
+         * Effected methods will return/accept paths in which only reserved characters are encoded. Methods returning paths will
+         * decode all non-reserved characters and remove only the <code>jsessionid</code> path parameter. All other path
+         * parameters will be returned and considered part of the path for the purposes of path matching. Methods that take
+         * paths expect the reserved characters to be encoded if they are to be considered part of a path segment, otherwise
+         * they will not be encoded and will have their reserved interpretation if possible. If a valid interpretation of a
+         * non-encoded reserved character is not possible, then the method will throw {@link MalformedURLException} if declared,
+         * otherwise a {@link IllegalArgumentException}.
+         */
+        ENCODED,
+    }
+
+    /**
+     * Return the path coding mode for the context. By default, this will be {@link PathCodingMode#LEGACY} if the
+     * {@link #getEffectiveMajorVersion()} less that or equal to 5, otherwise it will be {@link PathCodingMode#DECODED}. The
+     * default may be changed by container specific mechanisms or overridden by a call to
+     * {@link #setPathCodingMode(PathCodingMode)}
+     *
+     * @return The path coding mode for this context.
+     */
+    PathCodingMode getPathCodingMode();
+
+    /**
+     * Set the path coding mode for the context. This method may only be called
```

Comment:
Looks like this javadoc is incomplete

Reply:
yep - forgot to `mvn install` before a push.  Work in progress, so changes and reformats coming in.... but any early thoughts appreciated to make sure I'm barking at a tree that's at least in the right forest.